app/Http/Controllers/API/PageConfigController.php

```diff
@@ -7,119 +7,291 @@
 use Illuminate\Http\Request;
 use Illuminate\Http\JsonResponse;
 use App\Models\API\PageConfig;
+use App\Models\FileMapper;
 use Illuminate\Support\Facades\Auth;
 use Validator;
 use Log;
 
 class PageConfigController extends Controller
 {
 
-    public function store(Request $request): JsonResponse
-    {
-        $valRules = [
-            'page_type' => 'required|string',
-            'name' => 'required|string',
-            'description' => 'required|string',
-            'img_link' => 'required|string',
-            'parent' => 'required|integer',
-            'header_img' => 'required|string',
-            'header_text' => 'required|string',
-            'tenant_id' => 'required|integer',
-            'seq_no' => 'required|integer',
-            'language' => 'required|string'
-        ];
-        
-        $data = json_decode($request->getContent(), true);
-        $data['tenant_id'] = $request->header('tenant_id',0); 
-        $data['created_at'] = gmdate('Y-m-d H:i:s'); 
-        $data['updated_at'] = gmdate('Y-m-d H:i:s'); 
-
-        $validator = Validator::make($data, $valRules);
-
-        if (!$validator->passes()) {
-            dd($validator->errors()->all());
-        }
+   public function store(Request $request): JsonResponse
+{
+    $tenantId = $request->header('tenant_id');
+    $valRules = [
+        'page_type' => 'required|string',
+        'name' => 'required|string',
+        'description' => 'required|string',
+        'parent' => 'required|integer',
+        'header_img' => 'required|array',
+        'header_text' => 'required|string',
+        'seq_no' => 'required|integer',
+        'language' => 'required|string',
+        'page_url' => 'required|string',
+        'tenant_id' => 'required|integer|exists:tenants,id'
+    ];
+
+    $data = $request->all();
+    $data['tenant_id'] = $tenantId;
+    $data['created_at'] = gmdate('Y-m-d H:i:s');
+    $data['updated_at'] = gmdate('Y-m-d H:i:s');
 
-        $data['updated_by'] = Auth::user()->id;     
-       
-        $pageConfig = PageConfig::create($data);
+    $validator = Validator::make($data, $valRules);
 
+    if (!$validator->passes()) {
         return response()->json([
-            'data' => $pageConfig,
-            'message' => 'Success'
-        ], 200);
+            'success' => false,
+            'message' => 'Validation errors',
+            'errors' => $validator->errors()->all()
+        ], 422);
+    }
+
+    $data['updated_by'] = Auth::user()->id;
 
+    // Extract header_img from data
+    $headerImg = $data['header_img'];
+    unset($data['header_img']);
+
+    $pageConfig = PageConfig::create($data);
+
+    // Store header_img in FileMapper
+    foreach ($headerImg as $fileId) {
+        FileMapper::create([
```

Comment:
Use multi insert,  looping & inserting 1B1 is expensive operation and slow! 
check this example : 
https://stackoverflow.com/questions/29723865/how-to-insert-multiple-rows-from-a-single-query-using-eloquent-fluent